src/main/java/org/jboss/arquillian/extension/jacoco/client/ApplicationArchiveInstrumenter.java

```diff
@@ -47,6 +47,10 @@
    @Inject
    private Instance<JacocoConfiguration> config;
 
+   public void setConfig(Instance<JacocoConfiguration> config) {
+      this.config = config;
+   }
```

Comment:
Introducing setter just for the sake of testing shouldn't be a reason good enough to do so. I am aware that the main issue here is the fact that Arquillian's `@Inject` supports only field injection. If we would have support for constructor injection we could pass an instance of `config` instead. On the other hand, we can simply check if `config` is `null` before calling `.get()` on it and if so, provide some default value. Would that make sense?